packages/tx/src/transaction.ts

```diff
@@ -315,48 +280,114 @@ export default class Transaction {
   validate(stringError: true): string
   validate(stringError: boolean = false): boolean | string {
     const errors = []
+
     if (!this.verifySignature()) {
       errors.push('Invalid Signature')
     }
 
-    if (this.getBaseFee().cmp(new BN(this.gasLimit)) > 0) {
-      errors.push([`gas limit is too low. Need at least ${this.getBaseFee()}`])
+    if (this.getBaseFee().gt(this.gasLimit)) {
+      errors.push(`gasLimit is too low. given ${this.gasLimit}, need at least ${this.getBaseFee()}`)
     }
 
-    if (stringError === false) {
-      return errors.length === 0
-    } else {
+    if (stringError) {
       return errors.join(' ')
```

Comment:
This `join(' ')` has always looked out of place to me. I understand that `validate(true)` gives more info than just `validate()`, but why then combine that into a single string with an arbitrary format?

If we are introducing breaking changes here, we may want to change this too. Maybe returning `string[]`?

Reply:
👍 agree, done in https://github.com/ethereumjs/ethereumjs-vm/pull/812/commits/420d20ba988d3f5d2e4cf9317520fb981d2a038a

When I work on `block` I will have to remember to do the same to `block.validateTransactions(stringError)`